src/functTest/groovy/com/bmuschko/gradle/docker/tasks/image/DockerRemoveImageFunctionalTest.groovy

```diff
@@ -42,6 +42,75 @@ class DockerRemoveImageFunctionalTest extends AbstractGroovyDslFunctionalTest {
         !result.output.contains("repository")
     }
 
+    def "can remove multiple images"() {
+        // Given a few Official Images that are small... :-)
+        String firstImage = "alpine:3"
+        String secondImage = "busybox:1.36"
+        String thirdImage = "bash:5.2.21"
+
+        // And a build script that uses them...
+        buildFile << """
+            import com.bmuschko.gradle.docker.tasks.image.Dockerfile
+            import com.bmuschko.gradle.docker.tasks.image.DockerBuildImage
+            import com.bmuschko.gradle.docker.tasks.image.DockerListImages
+            import com.bmuschko.gradle.docker.tasks.image.DockerRemoveImage
+
+            task firstDockerfile(type: Dockerfile) {
+                from '$firstImage'
+                label(['maintainer': 'jane.doe@example.com'])
+            }
+            task buildFirstImage(type: DockerBuildImage) {
+                dependsOn firstDockerfile
+                inputDir = file("build/docker")
+            }
+
+            task secondDockerfile(type: Dockerfile) {
+                from '$secondImage'
+                label(['maintainer': 'jane.doe@example.com'])
+            }
+            task buildSecondImage(type: DockerBuildImage) {
+                dependsOn secondDockerfile
+                inputDir = file("build/docker")
+            }
+
+            task thirdDockerfile(type: Dockerfile) {
+                from '$thirdImage'
+                label(['maintainer': 'jane.doe@example.com'])
+            }
+            task buildThirdImage(type: DockerBuildImage) {
+                dependsOn thirdDockerfile
+                inputDir = file("build/docker")
+            }
+
+            task removeImages(type: DockerRemoveImage) {
+                dependsOn buildFirstImage
```

Comment:
We won't really need the `dependsOn` as you set an implicit dependency by assigning the image references (the output of `DockerBuildImage` tasks) to the input field `images`.